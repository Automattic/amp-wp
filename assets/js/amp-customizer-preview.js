( function( $ ) {
	'use strict';

	// Nav bar text color.
	wp.customize( 'amp_navbar_color', function( value ) {
		value.bind( function( to ) {
			$( '.amp-wp-header a' ).css( 'color', to );
			$( '.amp-wp-header div' ).css( 'color', to );
			$( '.amp-wp-header .amp-wp-site-icon' ).css( 'border-color', to ).css( 'background-color', to );
		} );
	} );

	// Nav bar background color.
	wp.customize( 'amp_navbar_background_color', function( value ) {
		value.bind( function( to ) {
			$( '.amp-wp-header' ).css( 'background-color', to );
			$( '.amp-wp-article a, .amp-wp-article a:visited, .amp-wp-footer a, .amp-wp-footer a:visited' ).css( 'color', to );
			$( 'blockquote, .amp-wp-byline amp-img' ).css( 'border-color', to );
		} );
	} );

	// Add custom-background-image body class when background image is added.
	wp.customize( 'amp_navbar_background_image', function( value ) {
		value.bind( function( newVal, oldVal ) {
			if ( newVal ) {
				$( '.amp-wp-header' ).addClass( 'header-background-image' ).css( 'background-image', 'url(' + newVal + ')' );
			} else {
				$( '.amp-wp-header' ).removeClass( 'header-background-image' ).css( 'background-image', 'none' );
			}
		} );
	} );

	// AMP background color scheme.
	wp.customize( 'amp_background_color', function( value ) {
		value.bind( function( to ) {

			// Light
			if ( to == 'light' ) {

				/*
					$theme_color      = '#fff';
					$text_color       = '#535353';
					$muted_text_color = '#9f9f9f';
					$border_color     = '#d4d4d4';
				*/

				$( 'body' ).css( 'background-color', '#ffffff' );
				$( 'body, a:hover, a:active, a:focus, blockquote, .amp-wp-article, .amp-wp-title, .amp-wp-comment-link a' ).css( 'color', '#535353' );
				$( '.amp-wp-meta, .wp-caption .wp-caption-text, .amp-wp-share-links, .amp-wp-tax-category, .amp-wp-tax-tag, .amp-wp-comment-link, .amp-wp-footer p' ).css( 'color', '#9f9f9f' );
				$( '.amp-wp-article-featured-image amp-img, .amp-wp-article-content amp-img, .wp-caption .wp-caption-text, .amp-wp-comment-link a, .adspace, .amp-wp-footer' ).css( 'border-color', '#d4d4d4' );
				$( '.amp-wp-iframe-placeholder, amp-carousel, amp-iframe, amp-youtube, amp-instagram, amp-vine' ).css( 'background-color', '#d4d4d4' );

			// Dark
			} else if ( to == 'dark' ) {

				/*
					$theme_color      = '#111';
					$text_color       = '#acacac';
					$muted_text_color = '#606060';
					$border_color     = '#2b2b2b';
				*/

				$( 'body' ).css( 'background-color', '#111111' );
				$( 'body, a:hover, a:active, a:focus, blockquote, .amp-wp-article, .amp-wp-title, .amp-wp-comment-link a' ).css( 'color', '#acacac' );
				$( '.amp-wp-meta, .wp-caption .wp-caption-text, .amp-wp-share-links, .amp-wp-tax-category, .amp-wp-tax-tag, .amp-wp-comment-link, .amp-wp-footer p' ).css( 'color', '#606060' );
				$( '.amp-wp-article-featured-image amp-img, .amp-wp-article-content amp-img, .wp-caption .wp-caption-text, .amp-wp-comment-link a, .adspace, .amp-wp-footer' ).css( 'border-color', '#2b2b2b' );
				$( '.amp-wp-iframe-placeholder, amp-carousel, amp-iframe, amp-youtube, amp-instagram, amp-vine' ).css( 'background-color', '#2b2b2b' );

			// Default
			} else {

				/*
					$theme_color      = '#fff';
					$text_color       = '#3d596d';
					$muted_text_color = '#87A6BC';
					$border_color     = '#c8d7e1';
				*/

				$( 'body' ).css( 'background-color', '#ffffff' );
				$( 'body, a:hover, a:active, a:focus, blockquote, .amp-wp-article, .amp-wp-title, .amp-wp-comment-link a' ).css( 'color', '#3d596d' );
				$( '.amp-wp-meta, .wp-caption .wp-caption-text, .amp-wp-share-links, .amp-wp-tax-category, .amp-wp-tax-tag, .amp-wp-comment-link, .amp-wp-footer p' ).css( 'color', '#87A6BC' );
				$( '.amp-wp-article-featured-image amp-img, .amp-wp-article-content amp-img, .wp-caption .wp-caption-text, .amp-wp-comment-link a, .adspace, .amp-wp-footer' ).css( 'border-color', '#c8d7e1' );
				$( '.amp-wp-iframe-placeholder, amp-carousel, amp-iframe, amp-youtube, amp-instagram, amp-vine' ).css( 'background-color', '#c8d7e1' );

			}
		} );
	} );

} )( jQuery );
