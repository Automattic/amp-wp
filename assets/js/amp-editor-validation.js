/*jshint esversion: 6 */
/*global amp, wp:true */
/**
 * AMP Gutenberg integration.
 *
 * On editing a block, this checks that the content is AMP-compatible.
 * And it displays a notice if it's not.
 */

/* exported ampGutenberg */
var ampEditorValidation = ( function( $ ) {
	'use strict';

	var component = {
		/**
		 * Holds data.
		 */
		data: {},

		/**
		 * The response from the amp.validator for valid AMP.
		 */
		validStatus: 'PASS',

		/**
		 * Boot module.
		 *
		 * @param {Object} data Object data.
		 * @return {void}
		 */
		boot: function( data ) {
			component.data = data;
			$( document ).ready( function() {
				component.getScript();
			} );
		},

		/**
		 * Gets the amp CDN validator.js script, and run this file.
		 *
		 * The getScript callback is a workaround, and not recommended for production.
		 * It replaces the wp value that the script overwrote.
		 *
		 * @returns void.
		 */
		getScript: function() {
			$.getScript( 'https://cdn.ampproject.org/v0/validator.js', function() {
				component.validatePage();
				if ( 'undefined' !== typeof wp.blocks ) {
					component.processBlocks();
				}
			} );
		},

		/**
		 * Gets all of the registered blocks, and overwrites their edit() functions.
		 *
		 * The new edit() functions will check if the content is AMP-compliant.
		 * If not, the block will display a notice.
		 *
		 * @returns {void}
		 */
		processBlocks: function() {
			var blocks = wp.blocks.getBlockTypes(),
				key = 'name';
			blocks.forEach( function( block ) {
				if ( block.hasOwnProperty( key ) ) {
					component.overwriteEdit( block[ key ] );
				}
			} );
		},

		/**
		 * Overwrites the edit() function of a block.
		 *
		 * Retain the original edit function in OriginalBlockEdit.
		 * If the block's content isn't valid AMP,
		 * Prepend a notice to the block.
		 *
		 * @see https://riad.blog/2017/10/16/one-thousand-and-one-way-to-extend-gutenberg-today/
		 * @param {string} blockType the type of the block, like 'core/paragraph'.
		 * @returns {void}
		 */
		overwriteEdit: function( blockType ) {
			var el = wp.element.createElement,
				Notice = wp.components.Notice,
				block = wp.blocks.unregisterBlockType( blockType ),
				OriginalBlockEdit = block.edit;

			block.edit = function( props ) {
				var result = [ el( OriginalBlockEdit, props ) ],
					content = block.save( props );

				if ( 'string' !== typeof content ) {
					content = wp.element.renderToString( content );
				}

				// If validation fails, prepend a Notice to the block.
				if ( ! component.isValidAMP( content, true ) ) {
					result.unshift( el(
						Notice,
						{
							status: 'warning',
							content: component.data.i18n.notice,
							isDismissible: false
						}
					) );
				}
				return result;
			};
			wp.blocks.registerBlockType( blockType, block );
		},

		/**
		 * Whether markup is valid AMP.
		 *
		 * Uses the AMP validator from the AMP CDN.
		 * And places the passed markup inside the <body> tag of a basic valid AMP page.
		 * Then, validates that page.
		 *
		 * @param {string} markup The markup to test.
		 * @param {boolean} doWrap Whether to wrap the passed markup in a basic AMP document.
		 * @returns {boolean} $valid Whether the passed markup is valid AMP.
		 */
		isValidAMP: function( markup, doWrap = false ) {
			var validated,
				validKey = 'status';
			if ( true === doWrap ) {
				markup = `<!doctype html><html âš¡><head><meta charset="utf-8"><link rel="canonical" href="./regular-html-version.html"><meta name="viewport" content="width=device-width,minimum-scale=1"><style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript><script async src="https://cdn.ampproject.org/v0.js"></script></head><body>${markup}</body></html>`;
			}
			validated = amp.validator.validateString( markup );
			return ( validated.hasOwnProperty( validKey ) && component.validStatus === validated[ validKey ] );
		},

		/**
		 * Validate the entire page that a URL produces.
		 *
		 * @returns {void}
		 */
		validatePage: function() {
			if ( ! component.data.hasOwnProperty( 'doValidatePage' ) || true !== component.data.doValidatePage ) {
				return;
			}

			$.get( component.data.permalink, function( data ) {
				if ( 'string' === typeof data && ! component.isValidAMP( data ) ) {
					let $notice = $( '<div>' )
						.addClass( 'notice notice-warning is-dismissible' )
						.append( $( '<p>' )
							.text( component.data.i18n.notice )
						);
					$( '.wp-header-end' ).after( $notice );
				}
			} );
		}
	};

	return component;

} )( window.jQuery );
