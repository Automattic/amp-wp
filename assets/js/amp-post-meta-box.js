/* exported ampPostMetaBox */

/**
 * AMP Post Meta Box.
 *
 * @since 0.6
 */
var ampPostMetaBox = ( function( $ ) {
	'use strict';

	// Exports.
	return {
		/**
		 * Holds data.
		 *
		 * @since 0.6
		 */
		data: {
			previewLink: '',
			disabled: false,
			statusInputName: ''
		},

		/**
		 * Toggle animation speed.
		 *
		 * @since 0.6
		 */
		toggleSpeed: 200,

		/**
		 * Core preview button selector.
		 *
		 * @since 0.6
		 */
		previewBtn: '#post-preview',

		/**
		 * AMP preview button selector.
		 *
		 * @since 0.6
		 */
		ampPreviewBtn: '#amp-post-preview',

		/**
		 * Boot plugin.
		 *
		 * @since 0.6
		 * @param {Object} data Object data.
		 * @return {void}
		 */
		boot: function( data ) {
			this.data = data;
			$( document ).ready( function() {
				if ( ! this.data.disabled ) {
					this.addPreviewButton();
				}
				this.listen();
			}.bind( this ) );
		},

		/**
		 * Events listener.
		 *
		 * @since 0.6
		 * @return {void}
		 */
		listen: function() {
			$( this.ampPreviewBtn ).on( 'click.amp-post-preview', function( e ) {
				e.preventDefault();
				this.onAmpPreviewButtonClick();
			}.bind( this ) );

			$( '.edit-amp-status, [href="#amp_status"]' ).click( function( e ) {
				e.preventDefault();
				this.toggleAmpStatus( $( e.target ) );
			}.bind( this ) );

			$( '#submitpost input[type="submit"]' ).on( 'click', function() {
				$( this.ampPreviewBtn ).addClass( 'amp-disabled' );
			}.bind( this ) );
		},

		/**
		 * Add AMP Preview button.
		 *
		 * @since 0.6
		 * @return {void}
		 */
		addPreviewButton: function() {
			var previewBtn = $( this.previewBtn );
			previewBtn.addClass( 'without-amp' );
			previewBtn
				.clone()
				.insertAfter( previewBtn )
				.prop( {
					'href': this.data.previewLink,
					'id': this.ampPreviewBtn.replace( '#', '' )
				} )
				.parent()
				.addClass( 'has-next-sibling' );
		},

		/**
		 * AMP Preview button click handler.
		 *
		 * We trigger the Core preview link for events propagation purposes.
		 *
		 * @since 0.6
		 * @return {void}
		 */
		onAmpPreviewButtonClick: function() {
			var $input;

			// Flag the AMP preview referer.
			$input = $( '<input>' )
				.prop( {
					'type': 'hidden',
					'name': 'amp-preview',
					'value': 'do-preview'
				} )
				.insertAfter( this.ampPreviewBtn );

			// Trigger Core preview button and remove AMP flag.
			$( this.previewBtn ).click();
			$input.remove();
		},

		/**
		 * Add AMP status toggle.
		 *
		 * @since 0.6
		 * @param {Object} $target Event target.
		 * @return {void}
		 */
		toggleAmpStatus: function( $target ) {
			var $container = $( '#amp-status-select' ),
				status = $container.data( 'amp-status' ),
				$checked,
				editAmpStatus = $( '.edit-amp-status' );

			// Don't modify status on cancel button click.
			if ( ! $target.hasClass( 'button-cancel' ) ) {
				status = $( '[name="' + this.data.statusInputName + '"]:checked' ).val();
			}

			$checked = $( '#amp-status-' + status );

			// Toggle elements.
			editAmpStatus.fadeToggle( this.toggleSpeed, function() {
				if ( editAmpStatus.is( ':visible' ) ) {
					editAmpStatus.focus();
				}
			} );
			$container.slideToggle( this.toggleSpeed );

			// Update status.
			$container.data( 'amp-status', status );
			$checked.prop( 'checked', true );
			$( '.amp-status-text' ).text( $checked.next().text() );
		}
	};
})( window.jQuery );
