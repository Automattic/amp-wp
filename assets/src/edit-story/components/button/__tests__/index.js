/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import 'jest-styled-components';

/**
 * Internal dependencies
 */
import theme from '../../../theme.js';
import * as buttons from '../index.js';
import UndoIcon from '../icon_undo.svg';
import RedoIcon from '../icon_redo.svg';

const testDesc = {
	Primary: 'contain a modified color, border-color and background-color',
	Secondary: 'contain a modified color, border-color and background-color',
	Outline: 'contain a modified color and border-color',
	Undo: 'contain the undo svg icon',
	Redo: 'contain the redo svg icon',
};

// Loop over the buttons and run specific tests for each.
for ( const button in buttons ) {
	if ( buttons.hasOwnProperty( button ) ) { // eslint-disable-line import/namespace
		const Component = buttons[ button ];

		const { container } = render( <Component theme={ theme } /> );
		const btn = container.firstChild;

		describe( 'button', () => {
			it( `should match snapshot and ${ testDesc[ button ] }`, () => {
				expect( btn ).toMatchSnapshot();

				switch ( button ) {
					case 'Primary':
						expect( btn ).toHaveStyleRule( 'border-color', theme.colors.action );
						expect( btn ).toHaveStyleRule( 'background-color', theme.colors.action );
						expect( btn ).toHaveStyleRule( 'color', theme.colors.fg.v1 );

						break;

					case 'Secondary':
						expect( btn ).toHaveStyleRule( 'border-color', theme.colors.fg.v1 );
						expect( btn ).toHaveStyleRule( 'background-color', theme.colors.fg.v3 );
						expect( btn ).toHaveStyleRule( 'color', theme.colors.bg.v5 );

						break;

					case 'Outline':
						expect( btn ).toHaveStyleRule( 'border-color', theme.colors.fg.v2 );
						expect( btn ).toHaveStyleRule( 'color', theme.colors.fg.v1 );

						break;

					case 'Undo':
						expect( btn.innerHTML ).toStrictEqual( render( <UndoIcon /> ).container.innerHTML );

						break;

					case 'Redo':
						expect( btn.innerHTML ).toStrictEqual( render( <RedoIcon /> ).container.innerHTML );
						break;

					default:
						break;
				}
			} );
		} );
	}
}
