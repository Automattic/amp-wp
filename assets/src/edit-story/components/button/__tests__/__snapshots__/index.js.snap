// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot and contain a modified color and border-color 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  background: transparent;
  display: block;
  min-width: 63px;
  line-height: 28px;
  height: 30px;
  padding: 0 10px;
  cursor: pointer;
  border-color: #E5E5E5;
  color: #FFFFFF;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0 svg {
  width: 1em;
}

<button
  class="c0"
/>
`;

exports[`button should match snapshot and contain a modified color, border-color and background-color 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  background: transparent;
  display: block;
  min-width: 63px;
  line-height: 28px;
  height: 30px;
  padding: 0 10px;
  cursor: pointer;
  border-color: #47A0F4;
  background-color: #47A0F4;
  color: #FFFFFF;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0 svg {
  width: 1em;
}

<button
  class="c0"
/>
`;

exports[`button should match snapshot and contain a modified color, border-color and background-color 2`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  background: transparent;
  display: block;
  min-width: 63px;
  line-height: 28px;
  height: 30px;
  padding: 0 10px;
  cursor: pointer;
  border-color: #FFFFFF;
  background-color: #D4D3D4;
  color: #575D65;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0 svg {
  width: 1em;
}

<button
  class="c0"
/>
`;

exports[`button should match snapshot and contain the redo svg icon 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  background: transparent;
  display: block;
  min-width: initial;
  line-height: 28px;
  height: 30px;
  padding: 0 10px;
  cursor: pointer;
  border-color: #E5E5E5;
  color: #FFFFFF;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0 svg {
  width: 1em;
}

<button
  class="c0"
>
  <svg
    viewBox="0 0 497.25 497.25"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M248.625 89.25V0l127.5 127.5-127.5 127.5V140.25c-84.15 0-153 68.85-153 153s68.85 153 153 153 153-68.85 153-153h51c0 112.2-91.8 204-204 204s-204-91.8-204-204 91.8-204 204-204z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`button should match snapshot and contain the undo svg icon 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  background: transparent;
  display: block;
  min-width: initial;
  line-height: 28px;
  height: 30px;
  padding: 0 10px;
  cursor: pointer;
  border-color: #E5E5E5;
  color: #FFFFFF;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0 svg {
  width: 1em;
}

<button
  class="c0"
>
  <svg
    viewBox="0 0 497.25 497.25"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M248.625 89.25V0l-127.5 127.5 127.5 127.5V140.25c84.15 0 153 68.85 153 153s-68.85 153-153 153-153-68.85-153-153h-51c0 112.2 91.8 204 204 204s204-91.8 204-204-91.8-204-204-204z"
      fill="currentColor"
    />
  </svg>
</button>
`;
